### PR TITLE
Ferris reorg

### PR DESCRIPTION
**Pull request by pierrechevalier83**
Add some structure to the firmware for the Ferris keyboard.

## Description

* Add some structure to the ferris folder (add a 0.1 subfolder) to adapt it to the multiple versions that exist or are planned.
* Add my personal keymap to the Ferris
* Add a subfolder for the sweep by @davidphilipbarr.
* Ferris is bigger than a prototype now, with tens of keyboards out there and commercial availability in the plans once 0.2 has been tested. Move it up from the handwired subfolder.

Note: For testing, I successfully flashed both keymaps (default and pierrec83) to my Ferris compact and to my Ferris sweep.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ferris/sweep/rules.mk [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/ferris/sweep/rules.mk:11]:
```suggestion
```

File: keyboards/ferris/sweep/rules.mk [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/ferris/sweep/rules.mk:33]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
UNICODE_ENABLE = yes        # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/ferris/sweep/rules.mk [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+DEBUG_ENABLE = no
+CUSTOM_MATRIX = no         # Use custom matrix code
```
Comment (fauxpark) [inline keyboards/ferris/sweep/rules.mk:36]:
```suggestion

```

File: keyboards/ferris/sweep/config.h [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xC2AB
+#define PRODUCT_ID      0x3939
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    DPB
+#define PRODUCT         Ferris sweep
+#define DESCRIPTION     A minimalistic 34 - keys split keyboard
```
Comment (fauxpark) [inline keyboards/ferris/sweep/config.h:28]:
```suggestion
```

File: keyboards/ferris/0.1/0.1.c [hunk 5]
```diff
@@ -14,4 +14,4 @@ GNU General Public License for more details.
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
-#include "ferris.h"
+#include "0.1.h"
```
Comment (fauxpark) [inline keyboards/ferris/0.1/0.1.c:17]: This should be renamed `0_1`

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ferris/sweep/config.h (hunk 4 above)
Comment (pierrechevalier83) [reply keyboards/ferris/sweep/config.h:28]: Fix indentation in sweep's config

File: keyboards/ferris/0.1/0.1.c (hunk 5 above)
Comment (pierrechevalier83) [reply keyboards/ferris/0.1/0.1.c]: Done here: https://github.com/qmk/qmk_firmware/pull/10564/commits/a9a773f6bc4661286c8a361ac6d8abaca413d6dd

**Comment by pierrechevalier83** [comment]
Thanks for the review, @fauxpark :) I think I've addressed all your comments :D

**Review by drashna — APPROVED** [review]
LGTM.

**Review by fauxpark — APPROVED** [review]

File: keyboards/ferris/sweep/config.h [hunk 6]
```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2018 ENDO Katsuhiro <ka2hiro@curlybracket.co.jp>
```
Comment (noroadsleft) [inline keyboards/ferris/sweep/config.h:2]: Shouldn't this be something like:

```suggestion
Copyright 2020 David Philip Barr @davidphilipbarr
```

File: keyboards/ferris/sweep/sweep.c [hunk 7]
```diff
@@ -0,0 +1,16 @@
+/* Copyright 2018 ENDO Katsuhiro <ka2hiro@curlybracket.co.jp>
```
Comment (noroadsleft) [inline keyboards/ferris/sweep/sweep.c:1]: Again here.

File: keyboards/ferris/sweep/sweep.h [hunk 8]
```diff
@@ -0,0 +1,47 @@
+/* Copyright 2018 ENDO Katsuhiro <ka2hiro@curlybracket.co.jp>
```
Comment (noroadsleft) [inline keyboards/ferris/sweep/sweep.h:1]: And here.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/ferris/sweep/config.h (hunk 6 above)
Comment (pierrechevalier83) [reply keyboards/ferris/sweep/config.h:2]: I don't know. I took that line from DPB's repo, and thought they kept this from whatever they based their work on, so kept it as is. I guess the most correct alternative would be to append their name, and then may as well append mine too 🙂

Comment (noroadsleft) [reply keyboards/ferris/sweep/config.h:2]: There's a [section in the docs](https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=copyright-blurb) that goes over this.

Correct solution here from what you've told me is that it should have both David's name and yours.

Comment (pierrechevalier83) [reply keyboards/ferris/sweep/config.h]: I guess it depends on what's considered trivial or not :) I've updated the PR to add DPB and myself :)

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
> Ferris is bigger than a prototype now, with tens of keyboards out there and commercial availability in the plans once 0.2 has been tested. Move it up from the handwired subfolder.

Do you happen to have an idea of where all the boards are? What's the potential for end users getting caught out by this?

**Comment by pierrechevalier83** [comment]
> where all the boards are? What's the potential for end users getting



> > Ferris is bigger than a prototype now, with tens of keyboards out there and commercial availability in the plans once 0.2 has been tested. Move it up from the handwired subfolder.
> 
> Do you happen to have an idea of where all the boards are? What's the potential for end users getting caught out by this?

Most of the users who got boards from me are following the #ferris channel in the 40% discord server, where I will communicate the change.
I will also update the documentation in my own Ferris repo for anyone else.

This change will never get easier than it is now 🙂

**Comment by noroadsleft** [comment]
Sounds good.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**